From a8236cbc545ff1447ccd56af3dfa3653963dd3a2 Mon Sep 17 00:00:00 2001
From: Christopher Meiklejohn <cmeiklejohn@berklee.edu>
Date: Wed, 10 Feb 2010 20:19:54 -0500
Subject: [PATCH] Modified the Authorize.net gateway module to return an authorization code with each transaction

---
 .../billing/gateways/authorize_net.rb              |    3 +++
 1 files changed, 3 insertions(+), 0 deletions(-)

diff --git a/lib/active_merchant/billing/gateways/authorize_net.rb b/lib/active_merchant/billing/gateways/authorize_net.rb
index aca415c..119cec0 100644
--- a/lib/active_merchant/billing/gateways/authorize_net.rb
+++ b/lib/active_merchant/billing/gateways/authorize_net.rb
@@ -39,6 +39,7 @@ module ActiveMerchant #:nodoc:
       APPROVED, DECLINED, ERROR, FRAUD_REVIEW = 1, 2, 3, 4
 
       RESPONSE_CODE, RESPONSE_REASON_CODE, RESPONSE_REASON_TEXT = 0, 2, 3
+      AUTHORIZATION_CODE = 4
       AVS_RESULT_CODE, TRANSACTION_ID, CARD_CODE_RESPONSE_CODE  = 5, 6, 38
 
       self.supported_countries = ['US']
@@ -272,10 +273,12 @@ module ActiveMerchant #:nodoc:
           :response_code => fields[RESPONSE_CODE].to_i,
           :response_reason_code => fields[RESPONSE_REASON_CODE], 
           :response_reason_text => fields[RESPONSE_REASON_TEXT],
+          :authorization_code => fields[AUTHORIZATION_CODE],
           :avs_result_code => fields[AVS_RESULT_CODE],
           :transaction_id => fields[TRANSACTION_ID],
           :card_code => fields[CARD_CODE_RESPONSE_CODE]
         }
+
         results
       end
 
-- 
1.5.4.3

